AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -437,6 +440,24 @@ - (void)setPlayerState:(ASVideoNodePlayerState)playerState
   _playerState = playerState;
 }
 
+- (void)setAssetURL:(NSURL *)assetURL
+{
+  if (!ASObjectIsEqual(assetURL, self.assetURL)) {
+     self.asset = [AVURLAsset assetWithURL:assetURL];
```

Comment:
Yes - setting the asset will invoke [self clearFetchedData] and [self setNeedsDataFetch];  cc @Adlai-Holler who proposed this approach.